server/routes/archiveGoal/index.ts

```diff
@@ -14,7 +15,7 @@ export default (router: Router, services: Services) => {
 
   router.use('/plan/:prisonNumber/goals/:goalReference/archive', [checkUserHasEditAuthority()])
   router.get('/plan/:prisonNumber/goals/:goalReference/archive', [
-    retrieveAllGoalsForPrisoner(services.educationAndWorkPlanService),
+    retrieveGoals(services.educationAndWorkPlanService, GoalStatusValue.ACTIVE),
```

Comment:
Similar changes in the other controller route definitions - rather than retrieve all prisoner goals, we'll just get the goals in the status relevant to the operation in question.
For example, here for the "archive goal" route, we only want to get the ACTIVE goals, because we know you can't archive a completed or already archived goal.

Basically the changes in the other route defs are the same kind of idea 👍